core/include/detray/core/detector.hpp

```diff
@@ -449,6 +450,21 @@ class detector {
     DETRAY_HOST
     auto resource() { return _resource; }
 
+    /** @return the vecmem jagged vector buffer for surface candidates */
+    // TODO: det.get_n_max_objects_per_volume() is way too many for
+    // candidates size allocation. With the local navigation, the size can be
+    // restricted to much smaller value
+    vecmem::data::jagged_vector_buffer<intersection> create_candidates_buffer(
```

Comment:
Why do we need this in the ```detector``` instead of the ```navigator```? The ```detector``` should not need to know about the memory handling of the ```navigator```

Reply:
I think you are right. yeah it should be in navigator...

Reply:
Its fixed now